cmd/controller/app/controller.go

```diff
@@ -81,7 +81,7 @@ func Run(opts *options.ControllerOptions, stopCh <-chan struct{}) {
 			log := log.WithValues("controller", n)
 
 			// only run a controller if it's been enabled
-			if !util.Contains(opts.EnabledControllers, n) {
+			if !util.Contains(opts.EnabledControllers, n) || util.Contains(opts.DisabledControllers, n) {
```

Comment:
It would be good to have this specifically be unit tested with a few combinations of enabled and disabled controllers. Could be a new func in `util` which is called from here:

```go
func IsDisabled(n string, enabledControllers []string, disabledControllers []string) bool
```

I don't think it's required, though.